include/sir/platform.h

```diff
@@ -53,6 +53,16 @@
 #  endif
 # endif
 
+# if defined(__has_attribute) && (defined(__clang__) || defined(__GNUC__))
+#  if __has_attribute(no_sanitize)
+#   define SANITIZE_SUPPRESS(str) __attribute__((no_sanitize(str)))
+#  else
```

Comment:
Seems like this could be simplified;

```c
#if (stuff)
# if (otherstuff)
#   define thing(str) fancy(str)
# endif
#endif

#if !defined(thing)
# define thing(str)
#endif
```